ui/test/run_ui.sh

```
@@ -0,0 +1,157 @@
+# Same shape as forge-kernel/test/native/run_native.sh: every source compiles to
+# a .o ONCE, then each test links against the whole object set (so a duplicate
+# symbol across modules surfaces at link time), both phases parallel under a
+# portable bash-3.2 job cap, with a portable per-test timeout so a HANG FAILS.
+#
+# Exit 0 iff: every header is self-contained, the missing-include preflights
+# pass, every source and test compiles warning-clean, and every test exits 0.
+# Override with CXX=g++ / JOBS=N / TEST_TIMEOUT=secs / ONLY=<substring>.
+# ─────────────────────────────────────────────────────────────────────────────
+set -uo pipefail
+ROOT="$(cd "$(dirname "${BASH_SOURCE[0]}")/../.." && pwd)"
+cd "$ROOT"
```

Shellcheck (0.11.0)

[warning] Use 'cd ... || exit' or 'cd ... || return' in case cd fails.

(SC2164)